{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c5f77964",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting tkcalendar\n",
      "  Using cached tkcalendar-1.6.1-py3-none-any.whl.metadata (22 kB)\n",
      "Collecting babel (from tkcalendar)\n",
      "  Using cached babel-2.17.0-py3-none-any.whl.metadata (2.0 kB)\n",
      "Using cached tkcalendar-1.6.1-py3-none-any.whl (40 kB)\n",
      "Using cached babel-2.17.0-py3-none-any.whl (10.2 MB)\n",
      "Installing collected packages: babel, tkcalendar\n",
      "\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   ---------------------------------------- 0/2 [babel]\n",
      "   -------------------- ------------------- 1/2 [tkcalendar]\n",
      "   -------------------- ------------------- 1/2 [tkcalendar]\n",
      "   ---------------------------------------- 2/2 [tkcalendar]\n",
      "\n",
      "Successfully installed babel-2.17.0 tkcalendar-1.6.1\n"
     ]
    }
   ],
   "source": [
    "!pip install tkcalendar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "27481e08",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 592, in get\n",
      "    return self._tk.getint(value)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected integer but got \"\"\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\DELL\\AppData\\Local\\Temp\\ipykernel_13188\\2247739643.py\", line 45, in Save_to_file\n",
      "    a = st_ID.get()\n",
      "        ^^^^^^^^^^^\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 594, in get\n",
      "    return int(self._tk.getdouble(value))\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected floating-point number but got \"\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 592, in get\n",
      "    return self._tk.getint(value)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected integer but got \"\"\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\DELL\\AppData\\Local\\Temp\\ipykernel_13188\\2247739643.py\", line 45, in Save_to_file\n",
      "    a = st_ID.get()\n",
      "        ^^^^^^^^^^^\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 594, in get\n",
      "    return int(self._tk.getdouble(value))\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected floating-point number but got \"\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 592, in get\n",
      "    return self._tk.getint(value)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected integer but got \"\"\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\DELL\\AppData\\Local\\Temp\\ipykernel_13188\\2247739643.py\", line 45, in Save_to_file\n",
      "    a = st_ID.get()\n",
      "        ^^^^^^^^^^^\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 594, in get\n",
      "    return int(self._tk.getdouble(value))\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected floating-point number but got \"\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 592, in get\n",
      "    return self._tk.getint(value)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected integer but got \"\"\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\DELL\\AppData\\Local\\Temp\\ipykernel_13188\\2247739643.py\", line 45, in Save_to_file\n",
      "    a = st_ID.get()\n",
      "        ^^^^^^^^^^^\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 594, in get\n",
      "    return int(self._tk.getdouble(value))\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected floating-point number but got \"\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 592, in get\n",
      "    return self._tk.getint(value)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected integer but got \"\"\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\DELL\\AppData\\Local\\Temp\\ipykernel_13188\\2247739643.py\", line 45, in Save_to_file\n",
      "    a = st_ID.get()\n",
      "        ^^^^^^^^^^^\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 594, in get\n",
      "    return int(self._tk.getdouble(value))\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected floating-point number but got \"\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 592, in get\n",
      "    return self._tk.getint(value)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected integer but got \"\"\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 1948, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\DELL\\AppData\\Local\\Temp\\ipykernel_13188\\2247739643.py\", line 45, in Save_to_file\n",
      "    a = st_ID.get()\n",
      "        ^^^^^^^^^^^\n",
      "  File \"c:\\Users\\DELL\\OneDrive\\Desktop\\Data Analyst course\\Python\\New folder\\venv\\Lib\\tkinter\\__init__.py\", line 594, in get\n",
      "    return int(self._tk.getdouble(value))\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "_tkinter.TclError: expected floating-point number but got \"\"\n"
     ]
    }
   ],
   "source": [
    "import csv\n",
    "from tkinter import *\n",
    "from tkcalendar import DateEntry\n",
    "root = Tk()\n",
    "root.title('My sample')\n",
    "root.geometry('400x400')\n",
    "\n",
    "Label(root, text='student_id').grid(row=0, column=0)\n",
    "st_ID = IntVar()\n",
    "Entry(root, textvariable=st_ID).grid(row=0, column=1)\n",
    "st_ID.set('')\n",
    "\n",
    "Label(root, text='student_name').grid(row=1, column=0)\n",
    "st_name = StringVar()\n",
    "\n",
    "Entry(root, textvariable=st_name).grid(row=1, column=1)\n",
    "st_name.set('')\n",
    "\n",
    "Label(root, text='Mark1').grid(row=2, column=0)\n",
    "Mark1 = IntVar()\n",
    "Entry(root, textvariable=Mark1).grid(row=2, column=1)\n",
    "Mark1.set('')\n",
    "\n",
    "Label(root, text='Mark2').grid(row=3, column=0)\n",
    "Mark2 = IntVar()\n",
    "Entry(root, textvariable=Mark2).grid(row=3, column=1)\n",
    "Mark2.set('')\n",
    "\n",
    "Label(root, text='Mark3').grid(row=4, column=0)\n",
    "Mark3 = IntVar()\n",
    "Entry(root, textvariable=Mark3).grid(row=4, column=1)\n",
    "Mark3.set('')\n",
    "\n",
    "Label(root,text=\"DOB\").grid(row=5, column=0)\n",
    "dob = StringVar()\n",
    "DateEntry(root,textvariable=dob).grid(row=5, column=1)\n",
    "\n",
    "Label(root, text=\"Gender\").grid(row=6, column= 0)\n",
    "stggender = IntVar()\n",
    "Radiobutton(root, text=\"Male\", variable=stggender, value=1).grid(row=6, column=1)\n",
    "Radiobutton(root, text=\"Female\", variable=stggender, value=2).grid(row=6, column=2)\n",
    "\n",
    "\n",
    "def Save_to_file():\n",
    "    a = st_ID.get()\n",
    "    b = st_name.get()\n",
    "    c = Mark1.get()\n",
    "    d = Mark2.get()\n",
    "    e = Mark3.get()\n",
    "    total = c+d+e\n",
    "    do = dob.get()\n",
    "    Avg = int(total / 3)\n",
    "    if stggender.get() == 1:\n",
    "        gender = \"Male\"\n",
    "    else:\n",
    "        gender = \"Female\"\n",
    "\n",
    "    print(a, b, c, d, e, do, total, Avg, gender)\n",
    "\n",
    "    with open('student_details.csv', mode='a', newline='') as file:\n",
    "        w = csv.writer(file)\n",
    "        w.writerow([a, b, c, d, e, total, Avg, do, gender])\n",
    "    root.destroy()\n",
    "\n",
    "\n",
    "Button(root, text=\"Submit\", command=Save_to_file).grid(row=7, column=1)\n",
    "root.mainloop()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98fb524b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import tkinter as tk\n",
    "from tkinter import messagebox\n",
    "\n",
    "# Function to add task\n",
    "def add_task():\n",
    "    task = task_entry.get()\n",
    "    if task != \"\":\n",
    "        task_listbox.insert(tk.END, task)\n",
    "        task_entry.delete(0, tk.END)\n",
    "    else:\n",
    "        messagebox.showwarning(\"Warning\", \"Please enter a task!\")\n",
    "\n",
    "# Function to delete selected task\n",
    "def delete_task():\n",
    "    try:\n",
    "        selected_task_index = task_listbox.curselection()[0]\n",
    "        task_listbox.delete(selected_task_index)\n",
    "    except IndexError:\n",
    "        messagebox.showwarning(\"Warning\", \"Please select a task to delete.\")\n",
    "\n",
    "# Create main window\n",
    "window = tk.Tk()\n",
    "window.title(\"To-Do List\")\n",
    "window.geometry(\"300x400\")\n",
    "\n",
    "# Entry box to type task\n",
    "task_entry = tk.Entry(window, width=25, font=('Arial', 14))\n",
    "task_entry.pack(pady=10)\n",
    "\n",
    "# Add Task Button\n",
    "add_button = tk.Button(window, text=\"Add Task\", width=15, command=add_task)\n",
    "add_button.pack(pady=5)\n",
    "\n",
    "# Listbox to display tasks\n",
    "task_listbox = tk.Listbox(window, width=30, height=10, font=('Arial', 12))\n",
    "task_listbox.pack(pady=10)\n",
    "\n",
    "# Delete Task Button\n",
    "delete_button = tk.Button(window, text=\"Delete Task\", width=15, command=delete_task)\n",
    "delete_button.pack(pady=5)\n",
    "\n",
    "# Run the window\n",
    "window.mainloop()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2d4779cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "from tkinter import *\n",
    "\n",
    "# Function to evaluate the expression\n",
    "def click(event):\n",
    "    button_text = event.widget.cget(\"text\")\n",
    "    if button_text == \"=\":\n",
    "        try:\n",
    "            result = str(eval(str(entry_var.get())))\n",
    "            entry_var.set(result)\n",
    "        except:\n",
    "            entry_var.set(\"Error\")\n",
    "    elif button_text == \"C\":\n",
    "        entry_var.set(\"\")\n",
    "    else:\n",
    "        entry_var.set(entry_var.get() + button_text)\n",
    "\n",
    "# Creating GUI window\n",
    "root = Tk()\n",
    "root.title(\"Calculator\")\n",
    "root.geometry(\"300x400\")\n",
    "\n",
    "entry_var = StringVar()\n",
    "entry = Entry(root, textvariable=entry_var, font=\"Arial 20\")\n",
    "entry.pack(fill=BOTH, ipadx=8, pady=10, padx=10)\n",
    "\n",
    "# Button labels\n",
    "buttons = [\n",
    "    [\"7\", \"8\", \"9\", \"+\"],\n",
    "    [\"4\", \"5\", \"6\", \"-\"],\n",
    "    [\"1\", \"2\", \"3\", \"*\"],\n",
    "    [\"C\", \"0\", \"=\", \"/\"]\n",
    "]\n",
    "\n",
    "# Creating buttons dynamically\n",
    "for row in buttons:\n",
    "    frame = Frame(root)\n",
    "    frame.pack(expand=True, fill=\"both\")\n",
    "    for label in row:\n",
    "        btn = Button(frame, text=label, font=\"Arial 18\", relief=RAISED)\n",
    "        btn.pack(side=LEFT, expand=True, fill=\"both\", padx=2, pady=2)\n",
    "        btn.bind(\"<Button-1>\", click)\n",
    "\n",
    "root.mainloop()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3be2605b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
